service/worker/scanner/README.md

```diff
@@ -0,0 +1,247 @@
+# Scanner / fixer / etc how-to
```

Comment:
this is ... kind of is a how-to, but then descends into a manifesto. I agree 100% with all your points, but I'd maybe separate out the descriptive part ("tihs is how it works") from the normative part ("this is how it should be") and dump the normative part into a design-proposal. 

Having the two together, enmeshed would be an utterly baffling experience for the casual reader.